### PR TITLE
Azure Event Hubs August Release

### PR DESCRIPTION
**Pull request by antkmsft**

### CODE REVIEW

**Review by LarryOsterman — APPROVED** [review]

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md
Comment (ahsonkhan) [inline sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md:7]: @LarryOsterman, we should summarize and advertise fhe highlights of this release (producer/consumer client), to incentivize customers to use the package and give feedback.

I don't think it's useful to just say initial release, as a first beta, especially as these things go to the release notes and blog.

Comment (antkmsft) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md:7]: FWIW, for all the packages I've seen, the first version entry reads "Initial release" (it is only that "release" is lowercased).

Comment (antkmsft) [inline sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md:7]:
```suggestion
Initial release.
```

Comment (ahsonkhan) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md:7]: Even if that's so, I don't think that's the most user-friendly practice.

Here's an example of a good first release changelog entry. It mentions the important client class, the header that customers should include, and the relevant namespace. Adding a single sentence with a link to the service this SDK is supporting would also help discovery for new customers:
https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md#new-features-1

Comment (LarryOsterman) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md:7]: Do you have similar examples from the other languages? FOr instance, I often see "Initial Release. Please see the README for information", which isn't significantly different from "Initial Release"

Comment (ahsonkhan) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md:7]: I don't think there's a good rationale to follow that practice even from other languages, just because people tend to copy-paste and follow suite.

Also, given C++ is source distributed, there is more value in the changelog having some useful hints for customers to know where to look.

Comment (ahsonkhan) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md:7]: > And for the blog, if we need it, we can work with Cala.

Having a meaningful changelog for the first release, also helps Cala with writing the blog. She can fairly easy just leverage the contents of the changelog to write-up a sentence, rather than us having to later help create the blog snippet.

Comment (LarryOsterman) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md]: I've edited Antons comment to add additional information about supported scenarios.

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md
Comment (LarryOsterman) [inline sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md:7]:
```suggestion
Initial release.
Supported scenarios: Sending events and receiving events.

For more information, see README.md.

```

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md
Comment (LarryOsterman) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md]: No idea why it didn't let me edit it, but the 3rd time was the charm.

Comment (LarryOsterman) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md]: And IMHO the blog post shouldn't have much more than "Initial release" for the eventhubs service.

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md
Comment (ahsonkhan) [inline sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md:10]: @antkmsft would you mind fixing the formatting/spacing for this section. Adding a suggestion becomes tricky here.

Comment (LarryOsterman) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md:10]: What are the formatting/spacing errors you would like to see fixed?

Comment (ahsonkhan) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md]: I updated it directly in this PR as a commit, @LarryOsterman. Added a bullet (for consistency) and removed the extra new lines.

Please take a look and if you prefer something changed, let me know.

Also, based on suggestion from @antkmsft offline, I removed the "refer to the README.md" sentence. It doesn't have a link and might be confusing for folks not reading it on GitHub.

**Comment by antkmsft** [comment]
Once vcpkg team merges our amqp PR (https://github.com/microsoft/vcpkg/pull/32973), it is safe to release eventhubs.
We can trigger the release even before they merge, but the newly created PR (eventhubs) in their repo will be failing until they merge AMQP.

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md
Comment (LarryOsterman) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md]: The readme file is present in the repo. It should be linked in the changelog.

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md
Comment (LarryOsterman) [inline sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md:8]:
```suggestion
- Supported scenarios: Sending events and receiving events.
See [README.md](https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/eventhubs/README.md) for more information on how to use the EventHubs client.
```

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md
Comment (LarryOsterman) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md]: Added the refer to README.md file back with a link to the file for those folks not reading it on GitHub.

Comment (antkmsft) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md]: Changelogs is also available via the Azure SDK blog (https://devblogs.microsoft.com/azure-sdk/azure-sdk-release-july-2023/), specifically https://azure.github.io/azure-sdk/releases/2023-07/cpp.html. Over there, it is not clear which README.md we are talking about, there is no hyperlink. And if you're browsing in this very repo, and get to CHANGELOG.md, it is likely that you already saw the README.md, it is even one directory level above the place where CHANGELOG.md lives (and it gets displayed automatically once you are in the eventhubs directory).

Comment (LarryOsterman) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md]: As I see it, there are two options here:
1) The changelog.md is where customers discover the eventhubs service and want to discover what functionality is available. In that case, the customer needs to know where to go to find out the rest of the details on how to use the service.
2) The readme.md is where customers discover the eventhubs service, in which case the changelog.md can just have text indicating that it is the initial release.

If the blog strips hyperlinks from the changelog, then that is a bug in the logic which creates the blog post and it should be fixed by the team creating the blog post - if there's a hyperlink in a changelog, it's likely to be important.

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md
Comment (ahsonkhan) [inline sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md:9]: Use the messaging-eventhubs package-specific link to the README:
```suggestion
See [README.md](https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/eventhubs/azure-messaging-eventhubs/README.md) for more information on how to use the EventHubs client.
```

**Review by ahsonkhan — APPROVED** [review]
Other than using the more descriptive link to the readme which has specific content, looks good.

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md
Comment (antkmsft) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md]: It's good now - it has the hyperlink (whether this one or the one that Ahson suggests - I'd leave it up to you).
My comment was a bit confusing, sorry - note the 2 minute difference between Larry's commit that adds hyperlink and his comment, and my comment. Basically, I started writing my comment explaining why we want the link before I saw any of these changes. I think we're good now (but having no mentions of README would've been good for me as well).

**Comment by antkmsft** [comment]
@LarryOsterman, once AMQP Beta2 (https://github.com/microsoft/vcpkg/pull/32973) gets merged to vcpkg, let's also update this line: https://github.com/Azure/azure-sdk-for-cpp/blob/fd073360cf54c796ab098ed42c26f837af15a28f/sdk/eventhubs/azure-messaging-eventhubs/vcpkg/vcpkg.json#L17

I just discovered recently: You can add that edit to this PR via web, even without cloning my branch: in the file comparison mode, click "..." on the right, and then the file will be opened for editing, and on the left you'll have the entire directory tree - you can navigate to any file, click the edit (pencil) on the top, edit, and commit to this PR.

File: sdk/eventhubs/azure-messaging-eventhubs/vcpkg.json
Comment (antkmsft) [inline sdk/eventhubs/azure-messaging-eventhubs/vcpkg.json:8]:
```suggestion
    "name": "azure-core-amqp-cpp",
```

**Comment by antkmsft** [comment]
@LarryOsterman, we need to update `version>=` in `sdk/eventhubs/azure-messaging-eventhubs/vcpkg/vcpkg.json`, NOT in `sdk/eventhubs/azure-messaging-eventhubs/vcpkg.json`.

**Comment by antkmsft** [comment]
I made the update.

**Pull request merged**
